### PR TITLE
[socket-mode] Pass an agent to WebSocket in a proper way

### PR DESCRIPTION
**Pull request by g12i**
###  Summary

Fix passing an agent to `WebSocket`. `this.clientOptions.agent` was wrongly spread to WebSocket options, which effectively mean it wasn't used. 

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/socket-mode/src/SocketModeClient.ts [hunk 1]
```diff
@@ -451,7 +451,7 @@ export class SocketModeClient extends EventEmitter {
     const options: WebSocket.ClientOptions = {
       perMessageDeflate: false,
       ...this.clientOptions.tls,
-      ...this.clientOptions.agent,
+      ...(this.clientOptions.agent !== undefined ? { agent: this.clientOptions.agent } : {}),
```
Comment (g12i) [inline packages/socket-mode/src/SocketModeClient.ts:454]: Could be simplified to, if we are allowing explicit `undefined`:
```suggestion
      agent: this.clientOptions.agent
```

**Comment by mwbrooks** [comment]
Hey @g12i, thanks for catching this issue and sending a patch! I've added a couple reviewers, so someone will pick up it up shortly to confirm the bug and review the work.

If you have some time, I think we should also consider adding a test to prevent regression. :)

**Comment by g12i** [comment]
Hey @mwbrooks - trust me this is the first thing I wanted to do :)

```
PS C:\Users\wgrze\Projects\node-slack-sdk\packages\socket-mode> npm run test

> @slack/socket-mode@1.0.0 test C:\Users\wgrze\Projects\node-slack-sdk\packages\socket-mode
> npm run build && echo "Tests are not implemented." && exit 0


> @slack/socket-mode@1.0.0 build C:\Users\wgrze\Projects\node-slack-sdk\packages\socket-mode
> npm run build:clean && tsc


> @slack/socket-mode@1.0.0 build:clean C:\Users\wgrze\Projects\node-slack-sdk\packages\socket-mode
> shx rm -rf ./dist

"Tests are not implemented."
```

I see `mocha` in some packages but it's not defined in root `package.json` and I don't know the project well enough to build test setup myself.

**Comment by mwbrooks** [comment]
@g12i oops, my bad! SocketMode is hot off the press and don't have specs written for it yet. We've been looking into a reliable way to run non-integration tests and will most likely draw inspiration from the SocketMode tests for Python and Java.

**Review by stevengill — APPROVED** [review]
Thanks for catching this!

**Pull request merged**